File: 10035/Main.java
import java.io.*;
import java.util.*;

public class Main {
   public static void main(String args[]) throws IOException {
      String line;
      StringBuilder answer = new StringBuilder();
      BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

      while ((line = in.readLine()) != null) {
         StringTokenizer tokenizer = new StringTokenizer(line);
         int a = Integer.parseInt(tokenizer.nextToken());
         int b = Integer.parseInt(tokenizer.nextToken());

         if (a == 0 && b == 0) break;

         int sum = a + b;
         int carry = 0;
         boolean isCarry = false;
         String[] str = new String[2];

         str[0] = Integer.toString(sum);
         str[1] = Integer.toString((a > b) ? a : b);

         int i = str[0].length() - 1;
         int j = str[1].length() - 1;

         while (j >= 0) {
            if (str[0].charAt(i) < str[1].charAt(j)) {
               isCarry = true;
               ++carry;
            } else if (str[0].charAt(i) == str[1].charAt(j) && isCarry) {
              ++carry;
            } else {
               isCarry = false;
            }
            --i;
            --j;
         }
         if (carry > 0) {
            answer.append(Integer.toString(carry));
            answer.append(" carry ");
            answer.append((carry == 1) ? "operation.\n" : "operations.\n");
         } else {
            answer.append("No carry operation.\n");
         }
      }
      System.out.print(answer);
   }
}

File: 10035/main.c
#include <stdio.h>

int main(int argc, char* argv[])
{
   char str[2][10];
   int i, j, carry;
   int a, b, is_carry, sum;

   while (scanf("%d%d", &a, &b) == 2) {
      if (a == 0 && b == 0) break;
      sum = a + b;
      i = sprintf(str[0], "%d", sum);
      j = sprintf(str[1], "%d", (a > b) ? a : b);
      --i, --j;
      carry = is_carry = 0;
      while (j >= 0) {
         if (str[0][i] < str[1][j]) {
            is_carry = 1;
            ++carry;
         } else if (str[0][i] == str[1][j] && is_carry == 1) {
            ++carry;
         } else {
            is_carry = 0;
         }
         --i, --j;
      }
      if (carry) {
         printf("%d carry %s.\n", carry, (carry == 1) ? "operation" : "operations");
      } else {
         printf("No carry operation.\n");
      }
   }

   return 0;
}
